### ISSUE TITLE
`stem: true` prevents `filter_by` from returning results

### ISSUE THREAD

**Issue by jasonbosco** [typesense/typesense#1759]
## Description

When I `filter_by` on a field that has `stem: true` enabled, it returns 0 results.

## Steps to reproduce

Start Typesense:

```bash
export TYPESENSE_API_KEY=xyz

mkdir "$(pwd)"/typesense-data

docker run -p 8108:8108 \
            -v"$(pwd)"/typesense-data:/data typesense/typesense:27.0.rc13 \
            --data-dir /data \
            --api-key=$TYPESENSE_API_KEY \
            --enable-cors
```

Create a collection with `stem: true` and index a few docs:

```bash
➜  ~ export TYPESENSE_API_KEY=xyz
➜  ~ curl "http://localhost:8108/debug" \
       -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}"

{"state":1,"version":"27.0.rc13"}%
➜  ~ curl "http://localhost:8108/collections" \
       -X POST \
       -H "Content-Type: application/json" \
       -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
       -d '{
         "name": "companies",
         "fields": [
           {"name": "keywords", "type": "string[]", "facet": true, "stem": true }
         ]
       }'

{"created_at":1716829365,"default_sorting_field":"","enable_nested_fields":false,"fields":[{"facet":true,"index":true,"infix":false,"locale":"","name":"keywords","optional":false,"sort":false,"stem":true,"type":"string[]"}],"name":"companies","num_documents":0,"symbols_to_index":[],"token_separators":[]}%
➜  ~ curl "http://localhost:8108/collections/companies/documents/import?action=create" \
        -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
        -H "Content-Type: text/plain" \
        -X POST \
        -d '{"id": "124","keywords": ["Restaurant"]}
            {"id": "125","keywords": ["Baking"]}'

{"success":true}
{"success":true}
```

Now do a search with `filter_by`, notice how zero results are returned:

```bash
➜  ~ curl "http://localhost:8108/multi_search" \
        -X POST \
        -H "Content-Type: application/json" \
        -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
        -d '{
          "searches": [
            {
              "collection": "companies",
              "q": "*",
              "filter_by": "keywords:=Baking"
            }
          ]
        }'
{"results":[{"facet_counts":[],"found":0,"hits":[],"out_of":2,"page":1,"request_params":{"collection_name":"companies","first_q":"*","per_page":10,"q":"*"},"search_cutoff":false,"search_time_ms":0}]}%
```

Now, if you set `stem: false` in a new collection and repeat the same search, then it works as expected:

```bash
➜  ~ curl "http://localhost:8108/collections" \
       -X POST \
       -H "Content-Type: application/json" \
       -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
       -d '{
         "name": "companies2",
         "fields": [
           {"name": "keywords", "type": "string[]", "facet": true, "stem": false }
         ]
       }'

{"created_at":1716829423,"default_sorting_field":"","enable_nested_fields":false,"fields":[{"facet":true,"index":true,"infix":false,"locale":"","name":"keywords","optional":false,"sort":false,"stem":false,"type":"string[]"}],"name":"companies2","num_documents":0,"symbols_to_index":[],"token_separators":[]}%
➜  ~ curl "http://localhost:8108/collections/companies2/documents/import?action=create" \
        -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
        -H "Content-Type: text/plain" \
        -X POST \
        -d '{"id": "124","keywords": ["Restaurant"]}
            {"id": "125","keywords": ["Baking"]}'

{"success":true}
{"success":true}%
➜  ~ curl "http://localhost:8108/multi_search" \
        -X POST \
        -H "Content-Type: application/json" \
        -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
        -d '{
          "searches": [
            {
              "collection": "companies2",
              "q": "*",
              "filter_by": "keywords:=Baking"
            }
          ]
        }'
{"results":[{"facet_counts":[],"found":1,"hits":[{"document":{"id":"125","keywords":["Baking"]},"highlight":{},"highlights":[]}],"out_of":2,"page":1,"request_params":{"collection_name":"companies2","first_q":"*","per_page":10,"q":"*"},"search_cutoff":false,"search_time_ms":0}]}%
```

## Metadata

**Typesense Version**: 27.0.rc13

**OS**: Docker / Ubuntu

**Comment by Pegusus** [comment]
Thanks for the quick response, will be waiting for the fix. Regards

**Comment by ozanarmagan** [comment]
@Pegusus fixed with #1776

**Comment by jasonbosco** [comment]
Fix is now available in `27.0.rc18`
